sway/layout.c

```diff
@@ -108,6 +108,50 @@ swayc_t *remove_child(swayc_t *child) {
 	return parent;
 }
 
+//TODO: Implement horizontal movement.
+//TODO: Implement move to a different workspace.
+void move_container(swayc_t *container,swayc_t* root,enum movement_direction direction){
+	sway_log(L_DEBUG, "Moved window");
+	swayc_t *temp;
+	int i;
+	uint clength = root->children->length;
+	//Rearrange
+	for (i = 0; i < clength; ++i) {
+		swayc_t *child = root->children->items[i];
+		if(child->handle == container->handle){
+			if(clength == 1){
+				//Only one container, meh.
+				break;
+			}
+			if(direction == MOVE_LEFT && i > 0){
+				temp = root->children->items[i-1];
+				root->children->items[i] = temp;
+				root->children->items[i-1] = container;
```

Comment:
Sorry if it was a little wierd, but my logic was basically it's just shuffling a list round rather than moving a window. Could get a bit wierder when you have a mix of vertical and horizontal windows but we will see then.